File: src/main/java/de/florianschmitt/service/AbstractPageableAdminService.java
package de.florianschmitt.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;

import java.io.Serializable;

public abstract class AbstractPageableAdminService<ENTITY extends Serializable, REPOSITORY extends PagingAndSortingRepository<ENTITY, Long>> extends AbstractAdminService<ENTITY, REPOSITORY> implements BasePageableAdminService<ENTITY> {

    @Autowired
    protected REPOSITORY repository;

    @Override
    public Page<ENTITY> findAll(Pageable pageable) {
        return repository.findAll(pageable);
    }
}

File: src/main/kotlin/de/florianschmitt/service/base/AbstractPageableAdminService.kt
package de.florianschmitt.service.base

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import java.io.Serializable

abstract class AbstractPageableAdminService<ENTITY : Serializable, REPOSITORY : PagingAndSortingRepository<ENTITY, Long>> : AbstractAdminService<ENTITY, REPOSITORY>(), BasePageableAdminService<ENTITY> {

    @Autowired
    final override lateinit var repository: REPOSITORY

    override fun findAll(pageable: Pageable) = repository.findAll(pageable)
}
